Vfs integration tests p2  update vars

benfleis opened the pull request:
create test contract w/ consistent variables

consolidate keywords/env into consistent test contract
update many tests: data/->${TEST_DATA_LOC}/ to run against AWS, Azure
for same tests, update other variable usage as well (eg TEST_DIR, __WORKING_DIRECTORY__)
also rename _slow.test -> .test_slow
also add README describing them

Commit:
throw if non-varchar key

Commit:
more descriptive error message

benfleis committed:
consolidate keywords/env into consistent test contract

also add README describing them
manage working dir + env explicitly in TestConfiguration

Mytherin committed:
Destroy TaskNotifier prior to calling FinishTask

Mytherin committed:
Avoid calling shared_from_this() and instead use the passed in ClientContext in buffered data

benfleis committed:
update many tests: data/->${TEST_DATA_LOC}/ to run against AWS, Azure

- for any tests already touched, update other variable usage as well (eg TEST_DIR, __WORKING_DIRECTORY__)
- update windows-incompatible var substitutions results with equivalent path split
  variants
- also rename _slow.test -> .test_slow

Mytherin committed:
Remove zip bombs

carlopi reviewed (approved):
Thanks!! LGTM.

Only open question I have is whether it would make sense to leave `--data-location` in, and feed that into the test_env, since it might be simpler to pass down (like, it's independent from the rest).

Good to go in any case once CI is green.

carlopi commented:
That would be also my line of action, get this in, and adding that back can be done as add-on.

benfleis committed:
update many tests: data/->${TEST_DATA_LOC}/ to run against AWS, Azure

- for any tests already touched, update other variable usage as well (eg TEST_DIR, __WORKING_DIRECTORY__)
- update windows-incompatible var substitutions results with equivalent path split
  variants
- also rename _slow.test -> .test_slow
